src/main/java/ru/geekbrains/VaolEr/dao/ProductDAO.java

```diff
@@ -0,0 +1,73 @@
+package ru.geekbrains.VaolEr.dao;
+
+
+import ru.geekbrains.VaolEr.model.Product;
+
+
+import javax.persistence.EntityManager;
+import javax.persistence.Query;
+import java.util.List;
+
+public class ProductDAO {
+
+
+    public static boolean create(EntityManager entityManager, Product product){
+        try{
+            //Create
+            entityManager.getTransaction().begin();
+            entityManager.persist(product);
+            entityManager.getTransaction().commit();
+            return true;
+        } catch (Exception e){
+            e.printStackTrace();
+            return false;
+        }
+    }
+
+     public static Product findById(EntityManager entityManager, Long id){
+        //Read
+        //entityManager.getTransaction().begin();
+        return entityManager.find(Product.class, id);
+    }
+
+
+    public static List<Product> findAll(EntityManager entityManager){
+        entityManager.getTransaction().begin();
+        List<Product> list = (List<Product>) entityManager.createQuery("SELECT p from Product p")
+                .getResultList();
+        entityManager.getTransaction().commit();
+        return list;
+    }
+
+    public static Product saveOrUpdate(EntityManager entityManager, Product product) {
+        //Update
+        entityManager.getTransaction().begin();
+        Product productFromDb = findById(entityManager, product.getId());
+        productFromDb.setName(product.getName());
+        productFromDb.setCost(product.getCost());
+        entityManager.merge(productFromDb);
+        entityManager.getTransaction().commit();
+        return findById(entityManager, product.getId());
+    }
+
+    public static boolean delete(EntityManager entityManager, Product product){
+        try {
+            entityManager.getTransaction().begin();
+            Product productFromDb = findById(entityManager, product.getId());
+            if(productFromDb.equals(null)){
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [EC: Call to equals(null) (EC_NULL_ARG)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#ec-call-to-equals-null-ec-null-arg).
This method calls equals(Object), passing a null value as the argument. According to the contract of the equals() method, this call should always return false.